packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -176,8 +174,15 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         }
 
         if (electionData.addressVoted[msg.sender]) {
-            // TODO: if msg.sender already voted, rollback previous votes and allow to re-vote
-            revert AlreadyVoted();
+            uint voteDataIndex = electionData.votesIndexes[msg.sender];
+            VoteData memory previousVoteData = electionData.votes[voteDataIndex];
+
+            for (uint i = 0; i < previousVoteData.candidates.length; i++) {
+                electionData.nomineeVotes[previousVoteData.candidates[i]] += previousVoteData.votePower;
```

Comment:
Shouldn't this have to substract (`-=`) the votePower instead of add (`+=`) it ?

Reply:
+1

Reply:
🤦‍♂️ yes. Thanks! Fixed